plugins/sslservices/sslservices.cpp

```diff
@@ -203,51 +204,42 @@ class PKeyCache
 
         if (pkey)
         {
-            cache.emplace_front(hashc(reinterpret_cast<const byte *>(passphrase), passphraseLen, hashc(reinterpret_cast<const byte *>(key), keyLen, 0)), pkey);
-            if (cache.size() > OPENSSL_MAX_CACHE_SIZE)
-            {
-                EVP_PKEY_free(std::get<1>(cache.back()));
+            cache.emplace_front(hashc(reinterpret_cast<const byte *>(passphrase), passphraseLen, hashc(reinterpret_cast<const byte *>(key), keyLen, 0)), std::move(std::unique_ptr<EVP_PKEY, decltype(&EVP_PKEY_free)>(pkey, EVP_PKEY_free)));
+            if (cache.size() > SSLSERVICES_MAX_CACHE_SIZE)
                 cache.pop_back();
-            }
         }
         else
             failOpenSSLError("loading a pkey");
 
         return pkey;
-    };
+    }
 
-    void clear()
-    {
-        for (auto& c : cache)
-            EVP_PKEY_free(std::get<1>(c));
-        cache.clear();
-    };
+    void clear() {cache.clear();}
 
-    void printStatistics() {DBGLOG("OPENSSL PKEY CACHE STATS: HITS = %d, MISSES = %d", hits, misses);};
+    void printStatistics() {DBGLOG("SSLSERVICES PKEY CACHE STATS: HITS = %d, MISSES = %d", hits, misses);}
 
 private:
-    int hits;
-    int misses;
-    std::list<std::tuple<unsigned, EVP_PKEY *>> cache;
+    size32_t hits;
+    size32_t misses;
+    std::list<std::tuple<size32_t, std::unique_ptr<EVP_PKEY, decltype(&EVP_PKEY_free)>>> cache;
```

Comment:
cleanup: you can use a typedef/using to declare a type for a unique Pkey, which simplifies the code when adding to the cache